modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeGradleIncludeResourceCheck.java

```diff
@@ -107,39 +104,25 @@ private String _formatDependencies(String fileName, String content)
 			"\n\nliferayOSGi {\n\texpandCompileInclude = true\n}");
 	}
 
-	private String[] _getIncludeResourceJars(String fileName)
+	private List<String> _getIncludeResourceJars(String fileName)
 		throws IOException {
 
-		String bndFileName = StringUtil.replace(
-			fileName, "build.gradle", "bnd.bnd");
-
-		File file = new File(bndFileName);
-
-		String bndContent = FileUtil.read(file);
-
-		if (bndContent == null) {
-			return null;
-		}
+		BNDSettings bndSettings = getBNDSettings(fileName);
 
-		Matcher matcher1 = _includeResourcePattern.matcher(bndContent);
-
-		if (!matcher1.find()) {
+		if (bndSettings == null) {
 			return null;
 		}
 
-		String includeResources = matcher1.group();
+		String bndSettingsContent = bndSettings.getContent();
 
-		String jarDependencies = StringUtil.removeSubstring(
-			includeResources, "Include-Resource:\\");
+		List<String> includeResourceJars = BNDSourceUtil.getDefinitionValues(
+			bndSettingsContent, "-includeresource");
 
-		jarDependencies = StringUtil.removeSubstring(jarDependencies, "\n");
-		jarDependencies = StringUtil.removeSubstring(jarDependencies, "\t");
+		includeResourceJars.addAll(
+			BNDSourceUtil.getDefinitionValues(
+				bndSettingsContent, "Include-Resource"));
 
-		return StringUtil.split(jarDependencies, "\\");
+		return includeResourceJars;
 	}
 
-	private static final Pattern _includeResourcePattern = Pattern.compile(
-		"^(-includeresource|Include-Resource):[\\s\\S]*?([^\\\\]\n|\\Z)",
-		Pattern.MULTILINE);
```

Comment:
@kevhlee, can't we propagate this change to the other check implemented by @nicolas-sss?

IMHO, a big Regex is complex to understand and, apparently, there are util classes ready to assist in SF code, as you've demonstrated here. :slightly_smiling_face:

What do you think?

Reply:
Hey @natocesarrego! Long time no see. You are totally right. We could apply the utility methods for `BNDIncludeResourceCheck.java` as well. I'm gonna push one more commit.